server/datastore/mysql/software.go

```diff
@@ -241,26 +241,31 @@ func listSoftwareDB(
 }
 
 func selectSoftwareSQL(hostID *uint, opts fleet.SoftwareListOptions) (string, []interface{}, error) {
```

Comment:
Have you checked we have tests with all the combinations of `opts.Team == nil`, `opts.Team != nil`, `hostID == nil` & `hostID != nil`?

Reply:
We do, they failed until I nailed this exact change :)